server/graphql-schema.js

```diff
@@ -1062,6 +1073,21 @@ const graphqlSchema = gql`
         findOrCreateAtVersion(input: AtVersionInput!): AtVersion!
     }
 
+    """
+    """
+    enum RequiredReportPhase {
+        IS_CANDIDATE
+        IS_RECOMMENDED
+    }
+
+    type RequiredReportOperations {
+        """
+        """
+        createRequiredReport: RequiredReport!
+        updateRequiredReport(atId: ID!, browserId: ID!): RequiredReport!
+        deleteRequiredReport: RequiredReport!
+    }
```

Comment:
@howard-e I just wanted to ping you in case you have feedback on this idea.

The API definitely makes sense given the UI we are building for, but when I look at it from a GraphQL-first perspective I feel like it requires a lot of separate round trips and multiple endpoints, and it's difficult for the client to say, "I want these set of combinations, now make it so."

So what I want to suggest is something like this:

```
mutation {
  setAtBrowserCombinations(
    candidate: [{ atId: 1, browserId: 2 }, { atId: 2, browserId: 2 }, { atId: 3, browserId: 3 }]
  )
}
```

This would only require one endpoint / resolver, and we wouldn't need to add a requiredReports type to the mutations. Given the way the UI works it wouldn't reduce the number of round trips, but I do think there's an argument to be made for making the API even more declarative than it already is.

I feel like the biggest downside of my suggestion is the fact that what we already implemented clearly works, and of course it is currently implemented.

Reply:
> @howard-e I just wanted to ping you in case you have feedback on this idea.
> 
> The API definitely makes sense given the UI we are building for, but when I look at it from a GraphQL-first perspective I feel like it requires a lot of separate round trips and multiple endpoints, and it's difficult for the client to say, "I want these set of combinations, now make it so."
> 
> So what I want to suggest is something like this:
> 
> ```
> mutation {
>   setAtBrowserCombinations(
>     candidate: [{ atId: 1, browserId: 2 }, { atId: 2, browserId: 2 }, { atId: 3, browserId: 3 }]
>   )
> }
> ```
> 
> This would only require one endpoint / resolver, and we wouldn't need to add a requiredReports type to the mutations. Given the way the UI works it wouldn't reduce the number of round trips, but I do think there's an argument to be made for making the API even more declarative than it already is.

I do think that this may be best for any future use of that entity so I'd be onboard with this suggest. Seems good to me.

> I feel like the biggest downside of my suggestion is the fact that what we already implemented clearly works, and of course it is currently implemented.

My preference would be If it can be addressed with the other feedback items here, then that would be great. Otherwise, let's immediately create a follow up enhancement task when this lands